war/src/test/js/widgets/config/tabbar.spec.js

```diff
@@ -1,31 +1,79 @@
-var jsTest = require("jenkins-js-test");
-
-require('./mocks');
+import fs from 'fs';
+import path from 'path';
+import $ from 'jquery';
+import jsTest from '@jenkins-cd/js-test';
+import { mockBehaviorShim } from './mocks';
+
+const htmlConfigTabbedContent = fs.readFileSync(
+    path.resolve(__dirname, './freestyle-config-tabbed.html'),
+    'utf8'
+);
+
+function getConfigTabbar() {
+    // eslint-disable-next-line no-undef
+    return require('../../../../main/js/config-tabbar');
+}
+
+function getConfigTabbarWidget() {
+    // eslint-disable-next-line no-undef
+    return require('../../../../main/js/widgets/config/tabbar');
+}
 
 describe("tabbar-spec tests", function () {
+    // Need to mock the utils/page module because we will hijack the scroll events
+    const mockPageUtils = jest.requireActual('../../../../main/js/util/page');
+
+    beforeEach(() => {
+        mockBehaviorShim();
+
+        jest.mock('../../../../main/js/util/page', () => ({
+            __esModule: true,
+            ...mockPageUtils,
+            default: {
+                ...mockPageUtils.default,
+                fireBottomStickerAdjustEvent: jest.fn(),
+            }
+        }));
+    });
+
+    afterEach(() => {
+        jest.resetAllMocks()
+    });
+
+    afterAll(() => {
+        // Should call resetModules on afterAll because the test "test section activation"
+        // will break if resetModules is called on afterEach.
+        jest.resetModules();
+    });
 
     it("- test section count", function (done) {
         jsTest.onPage(function() {
-            var tabbars = jsTest.requireSrcModule('config-tabbar.js');
-            var firstTableMetadata = tabbars.tabs[0];
+            document.documentElement.innerHTML = htmlConfigTabbedContent;
 
-            var jQD = require('jquery-detached');
-            var $ = jQD.getJQuery();
+            var tabbars = getConfigTabbar();
+            var firstTableMetadata = tabbars.tabs[0];
 
-            expect($('.section-header-row', firstTableMetadata.configTable).size()).toBe(4);
+            expect(firstTableMetadata.configTable.find('.section-header-row').length).toBe(4);
             expect(firstTableMetadata.sectionCount()).toBe(4);
             expect($('.tabBar .tab').size()).toBe(4);
 
-            expect(firstTableMetadata.sectionIds().toString())
-                .toBe('config_general,config__advanced_project_options,config__build_triggers,config__build');
+            expect(firstTableMetadata.sectionIds()).toEqual([
+                'config_general',
+                'config__advanced_project_options',
+                'config__build_triggers',
+                'config__build'
+            ])
 
             done();
-        }, 'widgets/config/freestyle-config-tabbed.html');
+        }, htmlConfigTabbedContent);
     });
 
+    // This test may be deterministic, as it breaks if jest.resetModules is called on afterEach
```

Comment:
did you mean not deterministic?
```suggestion
    // This test may not be deterministic, as it breaks if jest.resetModules is called on afterEach
```

Reply:
Yeah, exactly